dataland-batch-manager/src/main/kotlin/org/dataland/datalandbatchmanager/service/IsinDeltaBuilder.kt

```diff
@@ -0,0 +1,116 @@
+package org.dataland.datalandbatchmanager.service
+
+import com.fasterxml.jackson.dataformat.csv.CsvMapper
+import com.fasterxml.jackson.dataformat.csv.CsvSchema
+import com.fasterxml.jackson.module.kotlin.kotlinModule
+import org.slf4j.LoggerFactory
+import org.springframework.beans.factory.annotation.Value
+import org.springframework.stereotype.Component
+import java.io.File
+
+/**
+ * The class to create csv file containing updated LEI-ISIN mapping
+ */
+@Component
+class IsinDeltaBuilder(
+    @Value("\${dataland.dataland-batch-manager.mapping-file}") private val savedMappingFile: File,
+) {
+
+    private val logger = LoggerFactory.getLogger(javaClass)
+
+    /**
+     * Creates the delta file of the updated LEI-ISIN mapping file
+     * @param newMappingFile latest version of the LEI-ISIN mapping file
+     * @param oldMappingFile previously stored version of the LEI-ISIN mapping file, if one exists
+     */
+    fun createDeltaOfMappingFile(newMappingFile: File, oldMappingFile: File?): Map<String, String> {
+        val newMapping = parseCsvToGroupedMap(newMappingFile)
+        if (oldMappingFile == null) {
+            replaceOldMappingFile(newMappingFile)
+            return newMapping
+        }
+        val oldMapping = parseCsvToGroupedMap(oldMappingFile)
+        val deltaMapping = findLeisWithUpdatedIsin(newMapping, oldMapping)
+        replaceOldMappingFile(newMappingFile)
+        return deltaMapping
+    }
+
+    /**
+     * Compares the two LEI-ISIN maps and identifies LEIs that have new ISINs
+     * @param newMap the new LEI-ISIN map
+     * @param oldMap the old LEI-ISIN map
+     * @return map of changed LEI-ISINs
+     */
+    fun findLeisWithUpdatedIsin(newMap: Map<String, String>, oldMap: Map<String, String>): Map<String, String> {
+        val deltaMap = mutableMapOf<String, String>()
+
+        for ((lei, newIsins) in newMap) {
+            val oldIsins = oldMap[lei]
+
+            if (oldIsins == null || oldIsins != newIsins) {
```

Comment:
What if oldIsins and newIsins only differ in order? If they would both be Set<String> this would not be a problem. There is nothing speaking against implementing them like that.

Reply:
The csv file is ordered alphabetically so a mere reordering is unlikely. and even if that happens, nothing horrible happens.

Reply:
Good point.